### ISSUE TITLE
Latest Nightly Build Fails to Instantiate Models with Bus Access Connections

### ISSUE THREAD

**Issue by jshackleton** [osate/osate2#2362]
## Summary
In the current OSATE nightly build (as of at least 6/9/2020), models do not instantiate if the model includes a bus access connection.

## Expected and Current Behavior
Models with bus access connections are failing to instantiate.

## Steps to Reproduce
1. Instantiate model enclosed below.
2. Watch it fail.
3. Failure point should report in the line "con1: bus access proc.b0 <-> sysbus;"

```aadl
package BusAccessExample
public
	
	bus Cable									
	end Cable;
	
	bus implementation Cable.Impl
		properties
			Transmission_Time => [Fixed => 0ns .. 0ns; PerByte => 1us .. 1us;];
	end Cable.Impl;
	
	processor Proc
		features
			b0: requires bus access Cable;
	end Proc;
	
	processor implementation Proc.Impl
		properties
			Priority_Range => 1 .. 32;
	end Proc.Impl;

	process App
	end App;

	process implementation App.Impl
	end App.Impl;
	
	system ABC
	end ABC;
	
	system implementation ABC.Impl
		subcomponents
			proc: processor Proc.Impl;
			sysbus: bus Cable.Impl;
			app: process App.Impl;
		connections
			con1: bus access proc.b0 <-> sysbus;
		properties
			Actual_Processor_Binding => (reference (proc)) applies to app;
	end ABC.Impl;

end BusAccessExample;
```

## Environment
Eclipse 2019-12
* **OSATE Version**:
OSATE 2-2.8.0 v20200604-1922 (testing)
* **Operating System**: 
MacOSX 10.13.6

**Comment by lwrage** [comment]
I cannot reproduce the issue. It works for me in the development version and the latest nightly build (on Linux).

Do you get anything in the error log? Do you get a dialog after instantiation (assuming you didn't turn it off)?

**Comment by jshackleton** [comment]
I was afraid of that. It is occuring in our nightly maven builds that use the upstream OSATE. I will make a better attempt to re-create the error. Stay tuned.

**Comment by jshackleton** [comment]
OK. So I've played with this example some more with OSATE 2-2.8.0. The following from the example above is flagged as an AADL error:

`
    con1: bus access proc.b0 <-> sysbus;
`

But changing the connection to unidirectional is accepted by 2.8.0:

`
    con1: bus access sysbus -> proc.b0;
`

It appears that the feature requires bus access must have an associated provides bus access to be a bi-directional connection. This is a condition not flagged in 2.7.1 or earlier.

So my post has turned into a question: what is the legality rule being violated by the convention in the original example, defining a bi-directional connection with a requires bus access connected directly to a bus resource?

Thanks!

**Comment by lwrage** [comment]
Bug in fix for #2344

**Comment by jshackleton** [comment]
Closing duplicate. Will track in #2344.
